### PR TITLE
feat: add module for views [DA-2397]

### PR DESCRIPTION
**Pull request by olli-honest**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* Create a module for creating BigQuery views.
* Fix an issue with BigLake table module schema. The schema should only be set when using JSON or CSV tables, see [the provider docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table) for more details.
* Fix a permission issue in the BigLake table test.
* Fix `checkov` checks.
* [Linear ticket](https://linear.app/honestbank/issue/DA-2397/add-module-for-views-in-terraform-gcp-bigquery).

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-bigquery/32)

### CODE REVIEW

**Review by chawatphonr — APPROVED** [review]

File: modules/gcp_bigquery_view/inputs.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+variable "name" {
+  type        = string
+  description = "A view name for the resource. Changing this forces a new resource to be created."
+}
+
+variable "description" {
+  type        = string
+  description = "The field description."
```
Comment (sunilhonest) [inline modules/gcp_bigquery_view/inputs.tf:13]: This should be description of what the view does.

File: modules/gcp_bigquery_view/inputs.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+variable "query" {
+  type        = string
+  description = "The view query"
```
Comment (sunilhonest) [inline modules/gcp_bigquery_view/inputs.tf:18]:
```suggestion
  description = "The view query."
```

File: modules/gcp_bigquery_view/main.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+resource "google_bigquery_table" "view" {
+  #checkov:skip=CKV_GCP_121:Deletion protection is not needed for views.
+  #checkov:skip=CKV_GCP_80:Encryption at rest does not apply to views.
+  dataset_id          = var.dataset_id
+  table_id            = var.name
+  deletion_protection = false
```
Comment (sunilhonest) [inline modules/gcp_bigquery_view/main.tf:6]: deletion_protection should be a input variable.

**Review by sunilhonest — CHANGES_REQUESTED** [review]

File: modules/gcp_bigquery_view/main.tf (hunk 3 above)
Comment (olli-honest) [reply modules/gcp_bigquery_view/main.tf:6]: Why? Views do not contain any data so there is no data loss if the view gets dropped.

File: modules/gcp_bigquery_view/inputs.tf (hunk 1 above)
Comment (olli-honest) [reply modules/gcp_bigquery_view/inputs.tf:13]: Any suggestion how to word it? This is what is used in the [GCP terraform provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table#description) and what [we use for tables](https://github.com/honestbank/terraform-gcp-bigquery/blob/main/modules/gcp_bigquery_table/inputs.tf#L13).

File: modules/gcp_bigquery_view/main.tf (hunk 3 above)
Comment (sunilhonest) [reply modules/gcp_bigquery_view/main.tf:6]: It is not about the data in views. It is about if the view can be deleted without any restrictions.

File: modules/gcp_bigquery_view/inputs.tf (hunk 1 above)
Comment (sunilhonest) [reply modules/gcp_bigquery_view/inputs.tf:13]: `The description of what the view does` should be good.

**Review by sunilhonest — APPROVED** [review]

**Pull request merged**